portal-impl/src/META-INF/security-spring.xml

```diff
@@ -21,8 +21,6 @@
 	<bean class="com.liferay.portal.kernel.security.auth.DefaultFullNameValidator" id="com.liferay.portal.kernel.security.auth.DefaultFullNameValidator" />
 	<bean class="com.liferay.portal.kernel.security.auth.DefaultScreenNameGenerator" id="com.liferay.portal.kernel.security.auth.DefaultScreenNameGenerator" />
 	<bean class="com.liferay.portal.kernel.security.auth.DefaultScreenNameValidator" id="com.liferay.portal.kernel.security.auth.DefaultScreenNameValidator" />
-	<bean class="com.liferay.portal.kernel.security.auth.LiberalEmailAddressValidator" id="com.liferay.portal.kernel.security.auth.LiberalEmailAddressValidator" />
```

Comment:
@JanisZhang Do you mean that they are only used by HookHotDeployListener, so they don't need to be published as Spring beans?

Reply:
@dantewang yes.

Reply:
@JanisZhang It's something similar with the full name generator?

Reply:
@dantewang
Yes, these classes should be defined in portal properties `users.screen.name.validator`,` users.email.address.validator`
and override the default one by HookHotDeployListener.